### PR TITLE
Fix double call to RefreshRevisions after creating a tag. Fixes #4495

### LINKED ISSUE #4495: GitEx selects the top revision after creating a new tag
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
GitEx selects the most top revision after creating a new tag. Which is somewhat inconvenient, especially if you were a few screens below.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

* Open any repository
* Select any revision below the top 
* Create a new tag for it
* After "Create Tag" dialog was closed, GitEx selects the top revision.

**What is the expected behavior?**
The previously selected revision must remain the same after create tag dialog was closed.

**Environment you encounter the issue:**

- GitExtensions version: 2.51
- GIT version: 2.15.1.windows.2
- OS version: Windows 7 x64

**Did this work in previous version of GitExtensions (which)?**

I think, 2.50 worked correctly.

### PR DESCRIPTION
**Pull request by pmme**
Fixes #4495.

Changes proposed in this pull request:
 - Stopped RevisionGrid::CreateTagToolStripMenuItemClick calling RefreshRevisions itself as this will happen when the create tag form is disposed of. Running RefreshRevisions twice loses the last selected revision.
 - Made running FormCreateTag instead use DoActionOnRepo, which is now consistent with other forms (e.g. FormCreateBranch) and consistent with how it's run from the main menu.
 
What did I do to test the code and ensure quality:
 - created a tag and observed that the selected revision did not change after the create tag dialog was closed.
 - observed that the tag was created and appeared in GitExtensions UI.
 - ensured that a tag could be created on a different revision by changing the option "Create tag at this revision", yet the selected revision did not change after the create tag dialog was closed.

Has been tested on (remove any that don't apply):
 - GIT 2.16.2.windows.1
 - Windows 10

### CODE REVIEW

File: GitUI/UserControls/RevisionGrid.cs [hunk 1]
```diff
@@ -2321,13 +2321,12 @@ private void CreateTagToolStripMenuItemClick(object sender, EventArgs e)
...
 
-            using (var frm = new FormCreateTag(UICommands, LatestSelectedRevision))
-            {
-                if (frm.ShowDialog(this) == DialogResult.OK)
+            UICommands.DoActionOnRepo(() =>
                 {
-                    RefreshRevisions();
-                }
-            }
+                    var frm = new FormCreateTag(UICommands, LatestSelectedRevision);
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid.cs]: the form still needs to be disposed of

Comment (pmme) [reply GitUI/UserControls/RevisionGrid.cs]: Please compare to the create branch code that is a few lines below. I copied that to ensure code consistency with how previous contributors have written things.
Happy to make changes but I don't know what needs to change.

Comment (RussKie) [reply GitUI/UserControls/RevisionGrid.cs]: The original code was `using (var frm = new FormCreateTag(UICommands, LatestSelectedRevision))`, it should be retained

**Comment by pmme** [comment]
@RussKie Hope those changes are what was required. 
If so, did you want that squashed? 
And, if it were squashed and force-pushed, would that update this pull request?

I am curious...
If RevisionGrid::CreateTagToolStripMenuItemClick requires "using" does that mean that CreateNewBranchToolStripMenuItemClick is deficient in that it does not? 
By way of comparison, "using" is used in both GitUiCommands::StartCreateBranchDialog and StartCreateTagDialog.

**Comment by pmme** [comment]
Ahhh phooey, didn't notice that my [R] key was not working in that commit message. :(

**Comment by RussKie** [comment]
Thank you

**Comment by RussKie** [comment]
> I am curious...
If RevisionGrid::CreateTagToolStripMenuItemClick requires "using" does that mean that CreateNewBranchToolStripMenuItemClick is deficient in that it does not?

Quite likely so. `using` ensures the object is disposed of once it goes out of scope

**Pull request merged**